src/index.css

```diff
@@ -0,0 +1,131 @@
+@import url('https://fonts.googleapis.com/css2?family=Righteous&family=Russo+One&display=swap');
```

Comment:
- [ ] The desktop view is now perfectly usable, however, the mobile view needs some tweaking:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/d3401740-ef57-4ac2-a35d-7dc03a2c0130)

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/fd621412-ce38-43ef-892a-e3eef76fe213)

The right arrow does not appear on the screen to navigate the cars.

## Explanation

The root of the problem is this:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/a8cf0c66-4f47-4bc8-bccf-3bce6c4281ac)

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/533df660-5e43-44ce-9bd0-e7685e1f96a9)

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/c54d7dfa-7626-4cbf-b719-67562acd3956)


You have content that does not fit the screen.  Therefore, a horizontal scroll is present on the homepage, the delete page, and the Reserve list page.

## Suggested solutions and extra help

### Possible Solution 1. 

This is the dirty but easy solution.

In the mobile view only, just make your cars a flex column and get rid of the carousel.  Get rid of the arrow buttons too, and let the user scroll vertically to navigate your cars.

I tested this, it's cheap, it's easy, and it works.  However, you won't get rid of the horizontal scroll (but the content will fit).

### Possible Solution 2.

This one requires more CSS tweaking, but the solution is permanent, all your content will fit on ALL mobile screens, and the horizontal scrolling will be gone.

#### If you follow these steps, you will be done in less than 10 minutes:

_**The main idea here is to make the navbar (left) 20% and the content (right) 80%.  20 + 80 = 100%**_



- In `sidebar.jsx`, add this class to line 17:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/2ed0efa2-02d1-4ff2-8038-1ba1d16b5783)

This will make the navbar max 20% of the screen.

- Now, go to `index.css`, and remove this line so the elements fit in the navbar:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/b04cda95-8d37-4940-8a13-e7c23e240194)

The result of the last 2 steps should be this one:

 
![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/e5e6e254-3629-4a0a-808d-5cb767134fec)

As you can see, our main problem, the horizontal scroll is gone.

- Now let's fix the "Delete" section because that one is still overflowing.  The problem is that the whole component is being wrapped with a header class.

Go to `DeletePage.jsx`.

Wrap ALL the jsx with a new `div`, so the next step works:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/5b3027d9-c4ee-4005-92e5-3e75f0e7035b)

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/8637ad55-39ef-4c52-ae15-ba4eb0f752b1)

After this, remove this closing `div` which corresponds to the div with the `heading` class:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/dd8cc126-fd01-4bcb-b164-62a0a9e45ec1)

Put a new closing `div` here, so we can wrap the problematic `div` that has the header class:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/7881d81e-365c-4296-aa3b-82ab87347c40)

Boom, now it works:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/47b4ca05-e8dc-4d10-b0ef-d384e219e6ea)

Here is the whole fixed return statement in `DeletePage.jsx`, for your convenience:

```javascript
  return (
    <div>
      <div className="Heading">
        <h1 className="mainh">Delete a car</h1>
      </div>
      <div className="w-[80vw] mx-auto">
        <div>
          <Carousel
            showArrows
            showThumbs={false}
            showStatus={false}
            dynamicHeight={false}
            centerMode
            centerSlidePercentage={40}
            renderArrowPrev={(onClickHandler, hasPrev, label) => (
              <button
                type="button"
                onClick={onClickHandler}
                title={label}
                className={`absolute top-1/2 left-0 transform -translate-y-1/2 z-10 bg-gray-500 text-white px-4 py-2 rounded mt-4 $  {
                    hasPrev ? 'hover:bg-green-600' : ''
                  }`}
              >
                &lt;
              </button>
            )}
            renderArrowNext={(onClickHandler, hasNext, label) => (
              <button
                type="button"
                onClick={onClickHandler}
                title={label}
                className={`absolute top-1/2 right-0 transform -translate-y-1/2 z-10 bg-gray-500 text-white px-4 py-2 rounded mt-4   ${
                  hasNext ? 'hover:bg-green-600' : ''
                }`}
              >
                &gt;
              </button>
            )}
          >
            {carData.map((car, index) => (
              <div key={car.id}>
                {randomCars[index] && (
                <img
                  src={randomCars[index]}
                  alt={`Random Car Rental ${index + 1}`}
                  className="w-full md:w-1/2 object-cover"
                />
                )}
                <div className="details">
                  <IconsCar />
                  <h5 className="title">{car.name}</h5>
                  <p>{car.description}</p>
                  <button
                    type="button"
                    onClick={() => handleDeleteClick(car.id)}
                    className="bg-red-500 text-white px-4 py-2 rounded hover:bg-red-600"
                  >
                    Delete
                  </button>
                </div>
              </div>
            ))}
          </Carousel>
        </div>
      </div>
    </div>
  );
```

- Now, let's put the arrows out of the way of the cars.

In `carousel.jsx`, add these classes:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/36997933-e531-453f-85f0-19134cbaf56c)

For your convenience, here are them:


Left button:

```JAVASCRIPT
              <button
                type="button"
                onClick={onClickHandler}
                title={label}
                className={`absolute max-[600px]:top-[85%] sm:top-3/4 left-0 transform -translate-y-1/2 z-10 bg-gray-500 text-white px-4 py-2 rounded mt-4 ${
                  hasPrev ? 'hover:bg-green-600' : ''
                }`}
              >
                &lt;
              </button>
```

Right button:

```JAVASCRIPT
              <button
                type="button"
                onClick={onClickHandler}
                title={label}
                className={`absolute max-[600px]:top-[85%] sm:top-3/4 right-0 transform -translate-y-1/2 z-10 bg-gray-500 text-white px-4 py-2 rounded mt-4 ${
                  hasNext ? 'hover:bg-green-600' : ''
                }`}
              >
                &gt;
              </button>
```

Result:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/61b2c282-caa1-471f-aa49-6933faea1df4)

Now, the buttons are less in the way.

- In order to fix the buttons position in `DeletePage.jsx`, do EXACTLY the same:



- Now let's make the sidebar elements look prettier.

On sidebar.jsx, modify ALL the sidebar main links (the 6 of them) by replacing the `px-4` classes with `sm:px-4` like this:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/8b43d24b-31d1-4126-aaa6-f2ecea64c940)

That way, the `px-4` will only apply to the desktop view.

Result:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/3cb6d642-7bfc-4e04-927d-6eb079868ab8)

- Finally, let's solve the icons!

For the mobile view, go to `icons.jsx`.  We need to display "hidden" for the mobile view (My time was running out to arrange them manually, and for the mobile view it was not a requirement) and display them for the desktop view.  Modify the line 12 exactly as shown:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/afaaa23d-50e4-473e-a54c-5b36afcfa02a)


Result:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/40690e84-7eda-4266-a7a2-f5c401aa3e5d)


For the desktop view, let's fix them because they are now misaligned:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/39ca58a0-c0e3-42fd-8d99-32b1b66de91f)

Go to sidebar.jsx and on line 83, remove the `justify-end` as shown:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/1e1df15d-9bd5-4442-85dc-47486c86c3de)

After all these steps, your project should look like this:

Mobile:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/d0d4b714-3f45-4ecf-b8ae-2fdb1aaf0904)

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/e5b06c1f-8bc5-4f61-8ac1-664d4a9d6230)

Desktop:

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/60edec50-68e1-4e43-b6ae-ad810be7db8f)

![image](https://github.com/MasumaJaffery/Car-Rent_App/assets/57421823/db53e43e-515f-4b42-9d42-1aaf5ec17008)

## Conclusion

Since the solutions have been handed to you, it should be easy now.  Please, implement it so your project looks more professional, and to improve your chances in the final evaluation. 💪 

_note: The solutions and examples shown in this review were created using my own experience and effort, and this should not be deemed as the "official" solution to the matter in any way.  This is only one way to fix the problem, written in order to help you._